Formula/smartdns.rb

```diff
@@ -0,0 +1,49 @@
+class Smartdns < Formula
+  desc "Rule-based DNS server for fast IP resolution, DoT/DoH/DoQ supported"
+  homepage "https://github.com/mokeyish/smartdns-rs"
+  url "https://github.com/mokeyish/smartdns-rs/archive/refs/tags/0.3.1.tar.gz"
+  sha256 "3a11e1b1c6ffc33d1120b6cec516d19a90375be283c114771cf950e2b364215d"
+  license "GPL-3.0-or-later"
+  head "https://github.com/mokeyish/smartdns-rs.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "bind" => :test # for `dig`
+  end
+
+  def install
+    system "cargo", "b", "-r"
+    sbin.install "target/release/smartdns"
```

Comment:
Doesn't this work?
```suggestion
    system "cargo", "install", *std_cargo_args
```

Reply:
`system "cargo", "install", *std_cargo_args` not support to install to `sbin`

see https://github.com/rust-lang/cargo/issues/7598

Reply:
Does it need to be in sbin?

Reply:
Yes, it is a dns server similar to dnsmasq, but it supports DoH, DoT and more, and needs to be kept in sync with the Linux platform, so it is better to install it in sbin.

Reply:
- [dnsmasq](https://github.com/Homebrew/homebrew-core/blob/master/Formula/dnsmasq.rb) - sbin
- [dnscrypt-proxy](https://github.com/Homebrew/homebrew-core/blob/master/Formula/dnscrypt-proxy.rb) -sbin

Reply:
We can use `std_cargo_args` then move binaries into `sbin`.

Reply:
> We can use `std_cargo_args` then move binaries into `sbin`.

I don't quite understand how to do, can you be more specific? move from `bin` to `sbin`?

Reply:
> move from `bin` to `sbin`?

Yes.

Reply:
Can you tell me what commands to use? I'm really not familiar with ruby.

Reply:
```ruby
system "cargo", "install", *std_cargo_args
sbin.install bin/"foo", bin/"bar"
```

Reply:
> ```ruby
> system "cargo", "install", *std_cargo_args
> sbin.install bin/"foo", bin/"bar"
> ```

Dose it will remove `foo`,`bar` from `bin`?

Reply:
@carlocab Dose it will remove foo,bar from bin?

Reply:
I just tested it, it works.